### PR TITLE
Rename hcp-metrics-collector to consul-telemetry-collector

### PR DESCRIPTION
**Pull request by clly**
### Description

We originally named the collector service name as hcp-metrics-collector. Rename the service name to follow the intended project name.

Also renamed the `bind_socket_dir` config key to remove the `hcp` reference. The collector can be used without HCP so it made sense to generalize the name

### Testing & Reproduction steps
- Unit tests have been modified to ensure everything passes.
- Tested locally as follows: 
1. Run consul agent in dev mode: `bin/consul agent -dev`
2. Register services:
  - web service: `consul services register web.json`
    ```json
    {
    "service": {
        "name": "web",
        "port": 9191,
        "connect": {
            "sidecar_service": {
                "proxy": {
                    "config": {
                        "envoy_telemetry_collector_bind_socket_dir": "/Users/ashvitha.sridharan/tmp/"
                    }
                }
            }
        }
    }
}```

   - collector service: `consul services register collector.json`
  ```json
{    "service":{
        "name": "consul-telemetry-collector",
        "port": 9090,
        "connect": {
            "sidecar_service": {
            }
        }
    }
}
```
2. Run sidecars
- `./bin/consul connect envoy -sidecar-for web -- -l debug`
- `./bin/consul connect envoy -admin-bind localhost:19001 -sidecar-for consul-telemetry-collector -- -l debug`

3. Run a mock metric handler (grpc service) and see metrics flow:
4. Screenshot 2023-05-12 at 3 40 46 PM

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [x] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: agent/proxycfg/connect_proxy.go [hunk 1]
```diff
@@ -628,16 +628,17 @@ func (s *handlerConnectProxy) handleUpdate(ctx context.Context, u UpdateEvent, s
...
-	// HCPMetricsBindSocketDir is a string that configures the directory for a
+// metricsCollectorConfig represents the basic opaque config values for pushing telemetry to
+// a consul telemetry collector.
+type metricsCollectorConfig struct {
+	// MetricsCollectorBindSocketDir is a string that configures the directory for a
 	// unix socket where Envoy will forward metrics. These metrics get pushed to
-	// the HCP Metrics collector to show service mesh metrics on HCP.
-	HCPMetricsBindSocketDir string `mapstructure:"envoy_hcp_metrics_bind_socket_dir"`
+	// the Consul Telemetry collector.
+	MetricsCollectorBindSocketDir string `mapstructure:"envoy_metrics_collector_bind_socket_dir"`
```
Comment (Achooo) [inline agent/proxycfg/connect_proxy.go:637]: @clly metrics felt maybe too generic, so I added "collector". Final name:`envoy_metrics_collector_bind_socket_dir`

Comment (Achooo) [reply agent/proxycfg/connect_proxy.go:637]: @freddygv we also discussed just calling it `envoy_telemetry_collector_bind_socket_dir` and re-using the socket for logs etc. We weren't sure if that would be a good idea as that could be a breaking change in the future or confusing if we can't send logs to the same socket. Let us know if you have any thoughts here!

File: command/connect/envoy/bootstrap_config.go [hunk 2]
```diff
@@ -845,7 +845,7 @@ func appendHCPMetricsConfig(args *BootstrapTplArgs, hcpMetricsBindSocketDir stri
 		  "transport_api_version": "V3",
 		  "grpc_service": {
 			"envoy_grpc": {
-			  "cluster_name": "hcp_metrics_collector"
+			  "cluster_name": "metrics_collector"
```
Comment (freddygv) [inline command/connect/envoy/bootstrap_config.go:848]: How come the naming is different here from `consul-telemetry-collector`?

What if this was something like "consul_telemetry_collector_loopback"?

Comment (clly) [reply command/connect/envoy/bootstrap_config.go:848]: SGTM!

Comment (Achooo) [reply command/connect/envoy/bootstrap_config.go:848]: yep that's better, will change it

File: website/content/commands/connect/envoy.mdx [hunk 3]
```diff
@@ -75,8 +75,8 @@ Usage: `consul connect envoy [options] [-- pass-through options]`
   In cases where either assumption is violated this flag will prevent the
   command attempting to resolve config from the local agent.
 
-- `envoy_hcp_metrics_bind_socket_dir` - Specifies the directory where Envoy creates a unix socket. 
-  Envoy sends metrics to the socket so that HCP collectors can connect to collect them." 
+- `envoy_telemetry_collector_bind_socket_dir` - Specifies the directory where Envoy creates a unix socket.
```
Comment (jjtimmons) [inline website/content/commands/connect/envoy.mdx:78]: This should be in https://developer.hashicorp.com/consul/docs/connect/proxies/envoy#control-bootstrap-configuration-from-proxy-configuration

Right now it's in section for flags for `consul envoy connect`

**Review by jjtimmons — CHANGES_REQUESTED** [review]

File: website/content/docs/connect/proxies/envoy.mdx [hunk 4]
```diff
@@ -190,6 +190,10 @@ the [`sidecar_service`](/consul/docs/connect/registration/sidecar-service) block
 - `envoy_stats_flush_interval` - Configures Envoy's
   [`stats_flush_interval`](https://www.envoyproxy.io/docs/envoy/v1.17.2/api-v3/config/bootstrap/v3/bootstrap.proto#envoy-v3-api-field-config-bootstrap-v3-bootstrap-stats-flush-interval).
 
+- `envoy_telemetry_collector_bind_socket_dir` - Specifies the directory where Envoy creates a unix socket.
```
Comment (Achooo) [inline website/content/docs/connect/proxies/envoy.mdx:193]: This should have been the right place to have the docs, so the docs were moved from `website/content/commands/connect/envoy.mdx` to this file.

File: website/content/docs/connect/proxies/envoy.mdx [hunk 5]
```diff
@@ -190,6 +190,10 @@ the [`sidecar_service`](/consul/docs/connect/registration/sidecar-service) block
 - `envoy_stats_flush_interval` - Configures Envoy's
   [`stats_flush_interval`](https://www.envoyproxy.io/docs/envoy/v1.17.2/api-v3/config/bootstrap/v3/bootstrap.proto#envoy-v3-api-field-config-bootstrap-v3-bootstrap-stats-flush-interval).
 
+- `envoy_telemetry_collector_bind_socket_dir` - Specifies the directory where Envoy creates a unix socket. 
+  Envoy sends metrics to the socket so that a consul telemetry collector can connect to collect them."
```
[the remark this answers is not in the document]
Comment (jjtimmons) [reply website/content/docs/connect/proxies/envoy.mdx:194]: maybe "Envoy sends metrics to the socket through which a consul telemetry collector collects the metrics."

[the remark this answers is not in the document]
Comment (Achooo) [reply website/content/docs/connect/proxies/envoy.mdx:194]: Whoops good catch will update that!

File: website/content/docs/connect/proxies/envoy.mdx [hunk 6]
```diff
@@ -190,6 +190,10 @@ the [`sidecar_service`](/consul/docs/connect/registration/sidecar-service) block
 - `envoy_stats_flush_interval` - Configures Envoy's
   [`stats_flush_interval`](https://www.envoyproxy.io/docs/envoy/v1.17.2/api-v3/config/bootstrap/v3/bootstrap.proto#envoy-v3-api-field-config-bootstrap-v3-bootstrap-stats-flush-interval).
 
+- `envoy_telemetry_collector_bind_socket_dir` - Specifies the directory where Envoy creates a unix socket. 
+  Envoy sends metrics to the socket with which a consul telemetry collector can collect them.
+  The socket is not configured by default.
```
Comment (trujillo-adam) [inline website/content/docs/connect/proxies/envoy.mdx:195]:
```suggestion
- `envoy_telemetry_collector_bind_socket_dir` - Specifies the directory where Envoy creates a Unix socket. 
  Envoy sends metrics to the socket where a Consul telemetry collector can collect them.
  The socket is not configured by default.
```
If I'm not mistaken, Unix is a proper noun.

**Review by trujillo-adam — COMMENTED** [review]
Minor suggestions for the docs.

**Review by trujillo-adam — APPROVED** [review]
Docs LGTM!

**Review by freddygv — APPROVED** [review]
LGTM

**Review by curtbushko — APPROVED** [review]
LGTM!  

You are re-naming everything and nothing is jumping out at me.

**Review by jjtimmons — APPROVED** [review]

**Pull request merged**